packages/api/src/post-build.js

```diff
@@ -192,6 +236,30 @@ class PostBuildHandler {
         handleError(res, err);
       }
     });
+
+    this.app.post('/api/build-gh', async (req, res, next) => {
+      try {
+        const buildInfo = PostBuildHandler.cleanBuildInfo(req.body.metadata); // Different line. The metadata object is the same as addbuild, already validated
+
+        req.body.data = await PostBuildHandler.flattenTap(req.body.data);
+        const parsedRaw = await PostBuildHandler.parseRawOutput(req.body.data, req.body.type);
+        const testCases = PostBuildHandler.parseTestCases(parsedRaw, req.body.data);
+
+        const isValid = await validateGithub(req.body.metadata.token, decodeURIComponent(req.body.metadata.repoId));
+        if (!isValid) {
+          throw new UnauthorizedError('Must have valid Github Token to post build');
+        }
+
+        if (req.body.metadata.private) {
+          throw new UnauthorizedError('Flaky does not store tests for private repos');
+        }
```

Comment:
Can we move this up near the top to do as little work as possible if the repo is private?